### PR TITLE
v0.1.1

### PR DESCRIPTION
**Pull request by nickvido**
# New Features
* Added report export helper methods: `download_product_report`, `download_asset_version_report`
* Updated Findings and Software Components queries to return more fields
* Added optional severity and status filters to `get_findings`
* Added optional `count` parameter to `get_findings`
* Added examples: `download_reports`, `generate_products_csv`
* Updated examples: `compare_versions`, `get_findings`
* Added ability to upload all supported third party scan types
* Updated `get_findings` to be able to take a single `asset_version_id` or a list

# Non-Breaking Changes
* Uses updated upload methods, removes usage of deprecated methods

# Bug Fixes
* Fixed minor bug when ordering Findings

# Testing Notes ✅ 
* All integration / end-to-end tests ran successfully
 * Creation Tests
 ** Full create product with CycloneDX Upload
 ** Full Create product with Binary Analysis Upload
 * Test 0.0.5: `test_get_findings`, `test_get_products`, `test_get_product_asset_versions`, `test_get_software_components`, `test_search_sbom`
 * Test 0.0.6: `test_get_findings`, `test_get_software_components`
 * Test 0.0.7: `test_get_assets`, `test_create_new_asset_version_artifact_and_test_for_upload_fix`
 * Test 0.0.8: `test_download_sboms`
 * Test 0.1.0: `test_raise_exception_on_graphql_error`
 * Test 0.1.1: `test_get_findings_count`, `test_get_findings_by_status`, `test_download_reports`, `test_upload_all_report_types`, `test_get_findings_by_severity`, `test_get_findings_for_multiple_asset_versions`

### CODE REVIEW

File: examples/download_reports.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import datetime
+import finite_state_sdk
+import requests
+
+
+def custom_download_report(token, organization_context, report_type="CSV", report_subtype="ALL_FINDINGS", asset_version_id=None, output_filename=None):
```
Comment (KK7NZY) [inline examples/download_reports.py:6]: The report_subtype field and report_type currently utilizes a string value. Considering this, would it be advisable to switch to using enums for better consistency and maintainability?

File: examples/download_reports.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        # Open a local file in binary write mode and write the content to it
+        print("File downloaded successfully.")
+        with open(output_filename, 'wb') as file:
+            file.write(response.content)
+            print(f'Wrote file to {output_filename}')
+    else:
+        print("Failed to download the file. Status code:", response.status_code)
+
+
+def example_download_reports(token, organization_context):
```
Comment (KK7NZY) [inline examples/download_reports.py:34]: Similar to the previous comment regarding the use of enums in place of strings, applies here as well.

File: finite_state_sdk/queries.py [hunk 3]
```diff
@@ -377,26 +418,50 @@ def _create_GET_FINDINGS_VARIABLES(asset_version_id=None, category=None, cve_id=
...
+GET_FINDINGS_COUNT = {
+    "query": """
+query GetFindingsCount($filter: FindingFilter)
+{
+    _allFindingsMeta(filter: $filter) {
+        count
+    }
+}
+""",
+    "variables": lambda asset_version_id=None, category=None, cve_id=None, status=None, severity=None, limit=None: _create_GET_FINDINGS_VARIABLES(asset_version_id=asset_version_id, category=category, cve_id=cve_id, status=status, severity=severity, limit=limit, count=True)
```
Comment (KK7NZY) [inline finite_state_sdk/queries.py:440]: Consider separating the variable assignments from the lambda expression for better readability and maintainability.

```python
def get_findings_variables(asset_version_id=None, category=None, cve_id=None, status=None, severity=None, limit=None):
    return _create_GET_FINDINGS_VARIABLES(
        asset_version_id=asset_version_id,
        category=category,
        cve_id=cve_id,
        status=status,
        severity=severity,
        limit=limit,
        count=True
    )
```

File: finite_state_sdk/queries.py [hunk 4]
```diff
@@ -471,7 +550,7 @@ def _create_GET_FINDINGS_VARIABLES(asset_version_id=None, category=None, cve_id=
         __typename
     }
 }""",
-    "variables": lambda asset_version_id=None, category=None, cve_id=None: _create_GET_FINDINGS_VARIABLES(asset_version_id=asset_version_id, category=category, cve_id=cve_id)
+    "variables": lambda asset_version_id=None, category=None, cve_id=None, status=None, severity=None, limit=None: _create_GET_FINDINGS_VARIABLES(asset_version_id=asset_version_id, category=category, cve_id=cve_id, severity=severity, status=status, limit=limit)
```
Comment (KK7NZY) [inline finite_state_sdk/queries.py:553]: Similar to the previous comment, consider separating the lambda expression into its own function for improved readability and maintainability.

File: finite_state_sdk/queries.py [hunk 5]
```diff
@@ -627,6 +724,81 @@ def _create_LAUNCH_CYCLONEDX_EXPORT_VARIABLES(cdx_subtype, asset_version_id):
...
 
+def _create_LAUNCH_REPORT_EXPORT_MUTATION(asset_version_id=None, product_id=None, report_type=None, report_subtype=None):
+    if not asset_version_id and not product_id:
+        raise Exception("Must specify either asset_version_id or product_id")
+
+    if asset_version_id and product_id:
+        raise Exception("Cannot specify both asset_version_id and product_id")
+
+    if asset_version_id:
+        if report_type == "CSV":
```
Comment (KK7NZY) [inline finite_state_sdk/queries.py:735]: Please consider replacing strings with Enum values for improved code clarity and maintainability.

**Review by KK7NZY — APPROVED** [review]
...

File: examples/compare_versions.py [hunk 6]
```diff
@@ -26,6 +29,47 @@
 """
 
 
+def compare_cves(version1_findings, version2_findings):
+    cve_changes = []
+
+    version1_cve_ids = []
+    version2_cve_ids = [finding['cves'][0]['cveId'] for finding in version2_findings]
```
[the remark this answers is not in the document]
Comment (nickvido) [reply examples/compare_versions.py:36]: Good suggestion - there is probably a lot of error checking that we could do for the examples. I didn't get too in-depth to keep them as simple and short as possible

File: examples/download_reports.py (hunk 1 above)
Comment (nickvido) [reply examples/download_reports.py:6]: Definitely will consider this for the next release

File: examples/download_reports.py (hunk 2 above)
Comment (nickvido) [reply examples/download_reports.py:34]: Same here - will consider this for the next release

File: finite_state_sdk/queries.py (hunk 3 above)
Comment (nickvido) [reply finite_state_sdk/queries.py:440]: Also a good suggestion - we can get this in a future release to improve readability for all these functions

**Pull request merged**